### PR TITLE
Add hooks for filtering media.

### PR DESCRIPTION
**Pull request by ianki**

### CODE REVIEW

**Comment by ianki** [comment]
This change is to support the Media Sync Filter.

Let me know if these new hooks look reasonable, and if there are any other places that media paths should be filtered.  Thanks

**Comment by dae** [comment]
Sorry, this will need rebasing to merge into master.

Paths are relative to the media folder already, so you should not need a full path - eg in mediasrv:

```
    if file_is_excluded(path):
        path = f"hidden/{path}"

    return aqt.mw.col.media.dir(), path
```

Please avoid adding the current working directory to the audio file - it can break things when paths contain non-Latin text on Windows for example.

File: qt/aqt/gui_hooks.py [hunk 1]
```diff
@@ -11,3 +11,5 @@
 
 # You can find the definitions in ../tools/genhooks_gui.py
 from aqt.hooks_gen import *
+
+
```
Comment (dae) [inline qt/aqt/gui_hooks.py:15]: These could be removed; otherwise looks good thanks. Rebasing onto master again should fix the Windows tests not working

Comment (ianki) [reply qt/aqt/gui_hooks.py]: Done.

Comment (ianki) [reply qt/aqt/gui_hooks.py]: I'm still getting a failure on Windows it seems.

**Comment by dae** [comment]
Cheers, will merge this once I sort out the Windows CI issue (which is an issue with the setup, not a problem with your changes)

**Pull request merged**